numpy/typing/tests/data/reveal/array_constructors.pyi

```diff
@@ -20,6 +20,9 @@ i8: np.int64
 A: npt.NDArray[np.float64]
 B: SubClass[np.float64]
 C: list[int]
+D: SubClass[np.float64 | np.int64]
+E: list[float]
+F: list[bool]
```

Comment:
to fully check whether the overlapping overloads are all compatible, you could add an explicit test for something like `list[float] | list[int]`